flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/expressions/PlannerExpressionConverter.scala

```diff
@@ -71,15 +98,22 @@ class PlannerExpressionConverter private extends ApiExpressionVisitor[PlannerExp
             val extraNames = args
               .drop(2)
               .map(e => getValue[String](e))
-            Alias(args.head, name, extraNames)
+            if (extraNames.nonEmpty) {
+              TableAlias(args.head, name, extraNames)
```

Comment:
Why do we need the `TableAlias`? From the code, it seems `as` will be converted into a TableAlias if the extraNames are not empty. However, a table can only contain one column, i.e., the extra name can be empty. 
Perhaps use `Alias` is enough? What do you think?

Reply:
I introduced `TableAlias` to handle naming the fields of result of `TableFunctionCall`. You are right that in 90% of the cases alias should have only one name. The only exception is the mentioned `TableFunctionCall`. If we separate those cases the logic of `Alias` becomes simpler. Right now the validation & resolution of `Alias` introduces an implicit contract that it can not be called for alias with extra names.

Besides in a bit longer run we want to get rid of `PlannerExpression`s at all, so this is a temporary solution.

Reply:
I agree with Hequn. Maybe we can avoid this change.